src/UI/SelectList.elm

```diff
@@ -0,0 +1,232 @@
+        , Element.clipY
+        ]
+        [ searchFieldView cfg opt.searchField
+        , opt.itens
+            |> filterOptions opt.searchField
+            |> List.map (\obj -> itemView cfg prop (isSelected obj) obj)
+            |> Element.column
+                [ Border.widthEach { bottom = 0, left = 0, right = 0, top = 1 }
+                , Border.color Palette.gray.lightest
+                , Element.width fill
+                , Element.height fill
+                , Element.clipY
+                , Element.scrollbarY
+                ]
+        , actionBarView cfg opt.actionBar
+        ]
+
+
+
+-- Internal
+
+
+actionBarView : RenderConfig -> Maybe ( String, String -> Icon, msg ) -> Element msg
+actionBarView cfg actionBar =
+    case actionBar of
+        Just ( title, icon, onClick ) ->
+            Element.row
+                [ Element.width fill
+                , Element.paddingEach
+                    { bottom = 17
+                    , left = 20
+                    , right = 12
+                    , top = 17
+                    }
+                , Background.color Palette.primary.lightest
+                , Font.color Palette.primary.middle
+                ]
+                [ Text.body2 title
+                    |> Text.withColor Text.colorPrimary
+                    |> Text.toEl cfg
+                , icon title
+                    |> Icon.toEl cfg
+                    |> Element.el
+                        [ Element.pointer
+                        , ARIA.roleAttr ARIA.roleButton
+                        , ARIA.labelAttr title
+                        , Events.onClick onClick
+                        , Element.alignRight
+                        ]
+                ]
+
+        Nothing ->
+            Element.none
+
+
+searchFieldView : RenderConfig -> Maybe (SearchOptions object msg) -> Element msg
+searchFieldView cfg searchField =
+    case searchField of
+        Just { label, searchMsg, currentFilter } ->
+            Element.row
+                [ Element.width fill
+                , Element.padding 12
+                ]
+                [ currentFilter
+                    |> Maybe.map Tuple.first
+                    |> Maybe.withDefault ""
+                    |> TextField.search searchMsg label
+                    |> TextField.withLabelNotHidden True
+                    |> TextField.withWidth TextField.widthFull
+                    |> TextField.withPlaceholder label
+                    |> TextField.withIcon (Icon.search "Search")
+                    |> TextField.toEl cfg
+                ]
+
+        Nothing ->
+            Element.none
+
+
+itemView : RenderConfig -> Properties object msg -> Bool -> object -> Element msg
+itemView cfg { select, renderItem } isSelected obj =
+    let
+        selectedBg : List (Attribute msg) -> List (Attribute msg)
+        selectedBg =
+            Palette.primary.middle
+                |> Background.color
+                |> (::)
+
+        bgColor =
```

Comment:
So, I'm not a huge fan of point-free style, I think can cause a lot of confusion. mainly because you needed to add a type signature on top to explain what's going on, and we almost never use this inside of `let...in` block.

I think the name of the function is really misleading. a function called, when you think about a function called `bgColor` what it supposes to return?  I would think maybe an `Element.Color` or an `Element.Attribute Msg`, but will never think about returning a list of attributes that are affecting sizing, spacing and etc. and also, it is not select, this function is not even adding a background color.

my suggestion would be

```elm
        selectedBg =
            Background.color Palette.primary.middle
```

and then

```elm
Element.el
    [...attributes
    ]
    |> prependIf isSelected selectedBg

prependIf : Bool -> a -> List a -> List a
prependIf shouldAdd item items =
    if shouldAdd then
         item :: items
    else
         items
```

Ps. this `prependIf` could go in the helper module we discussed before

Reply:
I liked this `prependIf` a lot, doing it!